fn_grpc_interface/fn_grpc_interface/components/function_grpc.py

```diff
@@ -0,0 +1,181 @@
+
+    def __init__(self, opts):
+        """constructor provides access to the configuration options"""
+        super(FunctionComponent, self).__init__(opts)
+        self.options = opts.get("fn_grpc_interface", {})
+
+    @handler("reload")
+    def _reload(self, event, opts):
+        """Configuration options have changed, save new values"""
+        self.options = opts.get("fn_grpc_interface", {})
+
+    @function("function_grpc")
+    def _function_grpc_function(self, event, *args, **kwargs):
+        """Function: A function to communicate with GRPC Servers based on the Communication Methods Specified."""
+
+        try:
+            # Initialising grpc helper class
+            grpc_helper_obj = GrpcHelperClass(logger_object=log)
+
+            # Get the function parameters:
+            grpc_channel = kwargs.get("grpc_channel")  # text
+            grpc_function = kwargs.get("grpc_function")  # text
+            grpc_function_data = kwargs.get("grpc_function_data")  # text
+
+            # Supported Communication types
+            SUPPORTED_COMM_METHODS = ["unary"]
+
+            log.info("grpc_channel: %s", grpc_channel)
+            log.info("grpc_function: %s", grpc_function)
+            log.info("grpc_function_data: %s", grpc_function_data)
+
+            yield StatusMessage("Connecting to gRPC Server...!")
+
+            # Parsing the variable data from config file and resilient inputs
+
+            # validate the gRPC Function Parameter to make sure data in this `helloworld:SayHello(HelloRequest)`
+            match_found = re.match(r'\w+\:\w+\(\w+\)$', grpc_function, re.IGNORECASE)
+            if match_found:
+                log.info("Function input 'grpc_function' is valid")
+            else:
+                raise ValueError("The grpc_function function input is in an incorrect format. Excepted format: 'helloworld:SayHello(HelloRequest)'")
+            try:
+                _grpc_package_name = grpc_function.split(':')[0].strip()
+                _grpc_rpc_stub_method_name = grpc_function.split(':')[1].split("(")[0].strip()
+                _grpc_request_method_name = re.sub("\)", '', grpc_function.split(':')[1].split("(")[1])
+                _grpc_request_method_name = _grpc_request_method_name.strip()
+            except Exception as err:
+                raise ValueError("The grpc_function function input is in an incorrect format. Excepted format: 'helloworld:SayHello(HelloRequest)'.\nERROR: {0}".format(err))
+
+            # Parsing the service configuration data
+            service_config_data = self.options.get(_grpc_package_name)
+            if not service_config_data:
+                raise ValueError("No configurations found for the gRPC server '{0}' in the app.config file".format(_grpc_package_name))
+            else:
+                try:
+                    config_file_data = service_config_data.split(',')
+                    _grpc_secure_connection = config_file_data[1]
```

Comment:
* Check here if any of these are `"None"` and then convert to `None` so we do not have to do further checks in the code

Reply:
ok